src/did/src/sign_strategy.rs

```diff
@@ -197,7 +211,7 @@ impl TransactionSigner for ManagementCanisterSigner {
         }
 
         let address = IcSigner {}
-            .public_key(self.key_id, DerivationPath::new(vec![]))
+            .public_key(self.key_id, self.derivation_path.clone())
             .await
             .map_err(|e| EvmError::from(format!("failed to get address: {e}")))?;
         let address = H160::from_slice(&address);
```

Comment:
Based on what reported by @Yasir, this line should be replaced with:
```
        let address: H160 = IcSigner
            .pubkey_to_address(&public_key)
            .map_err(|e| {
                EvmError::from(format!(
                    "failed to convert public key to ethereum address: {e}"
                ))
            })?
            .into();
```

Reply:
Done